frontend/test/resolution/testTuples.cpp

```diff
@@ -846,6 +845,262 @@ static void testTupleGeneric() {
+    )"""";
+
+  auto xQt = resolveQualifiedTypeOfX(context, program);
+  assert(!guard.realizeErrors());
+  assert(xQt.kind() == QualifiedType::VAR);
+  assert(xQt.type()->isRealType());
+
+  // Get the type of the '(x, _)' tuple itself.
+  auto mod = parseModule(context, program);
+  auto& rr = resolveModule(context, mod->id());
+  auto astTup = mod->stmt(4)->toCall()->actual(0);
+  assert(astTup->isTuple());
+  auto& qtTup = rr.byAst(astTup).type();
+  assert(qtTup.kind() == QualifiedType::CONST_VAR);
+  assert(qtTup.type()->isTupleType());
+}
+
+static void test22() {
+  printf("%s\n", __FUNCTION__);
+  Context ctx;
+  auto context = &ctx;
+  ErrorGuard guard(context);
+
+  std::string program =
+    R""""(
+    proc foo() { return (1.0, 2); }
+    var (v, _) = foo();
+    var x = v;
+    )"""";
+
+  auto qt = resolveQualifiedTypeOfX(context, program);
+  assert(!guard.realizeErrors());
+  assert(qt.kind() == QualifiedType::VAR);
+  assert(qt.type()->isRealType());
+}
+
+static void test23() {
+  printf("%s\n", __FUNCTION__);
+  Context ctx;
+  auto context = &ctx;
+  ErrorGuard guard(context);
+
+  std::string program =
+    R""""(
+    record r {}
+    iter foo(): (int, r) { yield (0, new r()); }
+    for (x, y) in foo() do;
+    )"""";
+
+  auto m = resolveTypesOfVariables(context, program, { "x", "y" });
+  assert(!guard.realizeErrors());
+  assert(m["x"].kind() == QualifiedType::CONST_VAR);
+  assert(m["x"].type()->isIntType());
+  assert(m["y"].kind() == QualifiedType::CONST_VAR);
+  assert(m["y"].type()->isRecordType());
+
+  auto mod = parseModule(context, program);
+  auto& rr = resolveModule(context, mod->id());
+  auto astTup = mod->stmt(2)->toIndexableLoop()->index();
+  assert(astTup->isTupleDecl());
+  auto& qtTup = rr.byAst(astTup).type();
+  assert(qtTup.kind() == QualifiedType::CONST_VAR);
+  assert(qtTup.type()->isTupleType());
+}
+
+static void test24() {
+  printf("%s\n", __FUNCTION__);
+  Context ctx;
+  auto context = &ctx;
+  ErrorGuard guard(context);
+
+  std::string program =
+    R""""(
+    record r {}
+    iter foo(): (int, r) { yield (0, new r()); }
+    for (x, _) in foo() do;
+    )"""";
+
+  auto qt = resolveTypeOfVariable(context, program, "x");
+  assert(!guard.realizeErrors());
+  assert(qt.kind() == QualifiedType::CONST_VAR);
+  assert(qt.type()->isIntType());
+}
+
+// This is private issue #6382.
+static void test25() {
+  printf("%s\n", __FUNCTION__);
+  Context ctx;
+  auto context = &ctx;
+  ErrorGuard guard(context);
+
+  std::string program =
+    R""""(
+    iter myIter() {
+      yield (1, 2);
+      yield (3, 4);
+    }
+
+    for (i, j) in myIter() {
+      var z = i;
+    }
+    )"""";
+
+  auto m = resolveTypesOfVariables(context, program, { "i", "j", "z" });
+  assert(!guard.realizeErrors());
+  assert(m["i"].kind() == QualifiedType::CONST_VAR);
+  assert(m["i"].type()->isIntType());
+  assert(m["j"].kind() == QualifiedType::CONST_VAR);
+  assert(m["j"].type()->isIntType());
```

Comment:
Could you explain why these are const?

Reply:
They're const because they're const in production, if the RHS is not a reference or a product of a `zip()` expression it's usually const.